interpreter/src/key/internal.rs

```diff
@@ -249,10 +249,43 @@ impl Key {
         }
     }
 
+    #[must_use] pub fn parent(&self) -> Segment {
```

Comment:
Come up with a better naming schem for Keys/parts of keys. 
I am currently thinking something along the lines of
- KeyBuf
- Key
- SubScript